base/util.jl

```diff
@@ -436,6 +447,33 @@ function julia_exename()
     end
 end
 
+"""
+    default_julia_basecmd() :: Cmd
+
+Default `julia` executable and options.
+"""
+default_julia_basecmd() = `$(joinpath(Sys.BINDIR::String, julia_exename()))`
```

Comment:
Why the type assert on `Sys.BINDIR`?

Reply:
I took it from how `julia_cmd` determines the path to `julia` in the current master:

https://github.com/JuliaLang/julia/blob/a646c26d85d29e3353c386cb8d8562443cfd248f/base/util.jl#L403

My guess was this is because `Sys.BINDIR` is a global and may be set to non-`String` object.

I also noticed `Sys.__init__` had the same type assertion:

https://github.com/JuliaLang/julia/blob/a646c26d85d29e3353c386cb8d8562443cfd248f/base/sysinfo.jl#L109